src/VirtoCommerce.CustomerExportImportModule.Data/Services/CustomerDataExporter.cs

```diff
@@ -0,0 +1,95 @@
+        {
+            _customerExportPagedDataSourceFactory = customerExportPagedDataSourceFactory;
+            _exportWriterFactory = exportWriterFactory;
+            _platformOptions = platformOptions.Value;
+        }
+
+        public async Task ExportAsync(ExportDataRequest request, Action<ExportProgressInfo> progressCallback, ICancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var exportProgress = new ExportProgressInfo { ProcessedCount = 0, Description = "Export has started" };
+
+            var dataSource = _customerExportPagedDataSourceFactory.Create(ModuleConstants.Settings.PageSize, request);
+
+            exportProgress.TotalCount = await dataSource.GetTotalCountAsync();
+            progressCallback(exportProgress);
+
+            const string exportDescription = "{0} out of {1} have been exported.";
+
+            exportProgress.Description = "Fetching...";
+            progressCallback(exportProgress);
+
+            var contactsFilePath = GetExportFilePath("Contacts");
+            var contactExportWriter = _exportWriterFactory.Create<ExportableContact>(contactsFilePath, new ExportConfiguration());
+
+            var organizationFilePath = GetExportFilePath("Organizations");
+            var organizationExportWriter = _exportWriterFactory.Create<ExportableOrganization>(organizationFilePath, new ExportConfiguration());
+
+            try
+            {
+                while (await dataSource.FetchAsync())
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var contacts = dataSource.Items.Select(x => x as ExportableContact).Where(x => x != null).ToArray();
+
+                    contactExportWriter.WriteRecords(contacts);
+
+                    var organizations = dataSource.Items.Select(x => x as ExportableOrganization).Where(x => x != null)
+                        .ToArray();
+
+                    organizationExportWriter.WriteRecords(organizations);
+
+                    exportProgress.ProcessedCount += dataSource.Items.Length;
+                    exportProgress.Description = string.Format(exportDescription, exportProgress.ProcessedCount,
+                        exportProgress.TotalCount);
+                    progressCallback(exportProgress);
+                }
+
+                exportProgress.Description = "Export completed";
+                exportProgress.FileUrls = new[] { contactsFilePath, organizationFilePath };
+            }
+            finally
+            {
+                contactExportWriter.Dispose();
+                organizationExportWriter.Dispose();
+            }
+        }
+
+        private string GetExportFilePath(string entityName)
+        {
+            if (string.IsNullOrEmpty(_platformOptions.DefaultExportFolder))
+            {
+                throw new PlatformException($"{nameof(_platformOptions.DefaultExportFolder)} must be set.");
```

Comment:
Platform exception? Maybe ArgumentNullException?

Reply:
It is a platform option. It is not the argument.